lib/matplotlib/axes/_base.py

```diff
@@ -2673,10 +2673,13 @@ def set_xscale(self, value, **kwargs):
         # around zero
         if value.lower() == 'log' and 'nonposx' not in kwargs.keys():
             kwargs['nonposx'] = 'clip'
-        self.xaxis._set_scale(value, **kwargs)
-        self.autoscale_view(scaley=False)
-        self._update_transScale()
-        self.stale = True
+
+        g = self.get_shared_x_axes()
+        for ax in g.get_siblings(self):
```

Comment:
no, because this is reaching in and touching the inner bits of all of the axes axis, not coming through the `ax.set_xscale` method (which is where we would get recursion).  My first attempt did use recursion,  but I think this is simpler as we don't need the logic to prevent the infinite recursion.